test/controllers/orderitems_controller_test.rb

```diff
@@ -56,16 +59,49 @@
       must_redirect_to order_path(order_id)
     end
 
-    it "will give a 400 error with invalid params" do
+    it 'will give a 400 error with invalid params' do
       product_id = nil
       test_input = {
-        "product_id": product_id,
+        "product_id": product_id
       }
-      expect {
+      expect do
         post products_path, params: test_input
-      }.wont_change "Product.count"
+      end.wont_change 'Product.count'
 
       must_respond_with :redirect
     end
   end
+
+  describe 'ship' do
+    describe 'logged in users' do
+      it 'will change orderitems shipped status to true' do
+        perform_login
```

Comment:
nothing major but i think we can pull the perform _login before the 'describe "logged in users" do' and then we don't have to perform login each time.

Reply:
done!